audioipc/src/codec.rs

```diff
@@ -159,29 +162,22 @@ where
 
     fn encode(&mut self, item: Self::In, buf: &mut BytesMut) -> io::Result<()> {
```

Comment:
Is the performance bottleneck here? Is it related to memory allocations?

Reply:
There are speed ups in both `encode` and `decode`.  I noticed these while upgrading to bytes 1.1 (not included, needs more work) where I accidentally made serialization slow.  I didn't profile before/after (speed up numbers are from the included criterion benchmark), so I don't have a precise explanation for where the improvements are.  In `decode` it seems to be mainly from avoiding the split & freeze.  Same with `encode` (using the extra buffer to calculate the length prefix after encoding), plus switching from passing a `Writer` in favour of the extra buffer into `serialize_into`.